DFC.EventGridSubscriptions.ApiFunction.UnitTests/DFC.EventGridSubscriptions.Services.Tests/SubscriptionRegistrationServiceTests.cs

```diff
@@ -18,15 +24,44 @@ namespace DFC.EventGridSubscriptions.ApiFunction.UnitTests.DFC.EventGridSubscrip
     public class SubscriptionRegistrationServiceTests
     {
         private readonly IOptionsMonitor<EventGridSubscriptionClientOptions> fakeClientOptions = A.Fake<IOptionsMonitor<EventGridSubscriptionClientOptions>>();
-        private readonly ILogger<SubscriptionRegistrationService> fakeLogger = A.Fake<ILogger<SubscriptionRegistrationService>>();
+        private readonly ILogger<SubscriptionService> fakeLogger = A.Fake<ILogger<SubscriptionService>>();
         private readonly IEventGridManagementClientWrapper fakeClient = A.Fake<IEventGridManagementClientWrapper>();
+        private readonly IDocumentService<SubscriptionModel> fakeDocumentClient = A.Fake<IDocumentService<SubscriptionModel>>();
+
+        public SubscriptionRegistrationServiceTests()
+        {
+            A.CallTo(() => fakeClientOptions.CurrentValue).Returns(new EventGridSubscriptionClientOptions { DeadLetterBlobContainerName = "DeadLetterContainer", StaleSubsriptionInterval = new TimeSpan(0, 0, 0, 10), StaleSubsriptionThreshold = 2 });
+        }
 
         [Fact]
         public async Task SubscriptionRegistrationServiceWhenAddSubscriptionAddsSubscription()
         {
             //Arrange
             A.CallTo(() => fakeClient.Topic_GetAsync(A<string>.Ignored, A<string>.Ignored, A<CancellationToken>.Ignored)).Returns(new Topic("location", "someid", "sometopic"));
-            var serviceToTest = new SubscriptionRegistrationService(fakeClientOptions, fakeClient, fakeLogger);
+
+            var serviceToTest = new SubscriptionService(fakeClientOptions, fakeClient, A.Fake<IDocumentService<SubscriptionModel>>(), fakeLogger);
+
+            //Act
+            var result = await serviceToTest.AddSubscription(new SubscriptionSettings { Endpoint = new Uri("http://somehost.com/awebhook"), Name = "Test Subscriber" });
+
+            //Assert
+            Assert.Equal(HttpStatusCode.Created, result);
+            A.CallTo(() => fakeClient.Topic_GetAsync(A<string>.Ignored, A<string>.Ignored, A<CancellationToken>.Ignored)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => fakeClient.Subscription_CreateOrUpdateAsync(A<string>.Ignored, A<string>.Ignored, A<EventSubscription>.Ignored, A<CancellationToken>.Ignored)).MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task SubscriptionRegistrationServiceWhenAddSubscriptionNullDocumentAddsSubscription()
+        {
+            //Arrange
+            A.CallTo(() => fakeClient.Topic_GetAsync(A<string>.Ignored, A<string>.Ignored, A<CancellationToken>.Ignored)).Returns(new Topic("location", "someid", "sometopic"));
+
+#nullable enable
```

Comment:
Could we enable nullable at the project level?

Reply:
I tried that, but then it breaks a lot of the tests where we pass a null parameter